caas/kubernetes/provider/specs/v3_test.go

```diff
@@ -172,7 +174,9 @@ kubernetesResources:
   serviceAccounts:
   - name: k8sServiceAccount1
     automountServiceAccountToken: true
-    global: true
+    roles: ["k8sRole", "k8sClusterRole"]
+  serviceAccountRoles:
```

Comment:
We designed the juju representation to remove the k8s boilerplate by embedding the rules in with the service account itself, rather than needing to do a reference to an external entity

Reply:
moved roles under service account now.